src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item/ProteinsListItem.tsx

```diff
@@ -60,22 +61,21 @@ const ProteinsListItem = (props: Props) => {
       });
       dispatch(replace(url));
     }
-    props.toggleExpandedProtein(transcript.id);
+    props.toggleExpandedProtein(product.stable_id);
   };
 
   const midStyles = classNames(transcriptsListStyles.middle, styles.middle);
 
   const itemRef = useRef<HTMLDivElement | null>(null);
   useEffect(() => {
-    if (transcript.id === transcriptIdToFocus) {
+    if (product.stable_id === transcriptIdToFocus) {
       setTimeout(() => {
         itemRef.current?.scrollIntoView({
           behavior: 'smooth'
         });
       }, 100);
-
       if (!props.expandedTranscriptIds.includes(transcriptIdToFocus)) {
-        props.toggleExpandedProtein(transcript.id);
+        props.toggleExpandedProtein(product.stable_id);
```

Comment:
Here, again, something that the code calls transcript id is compared to protein id.